core/ClassInfo.php

```diff
@@ -121,23 +131,31 @@ public static function baseDataClass($class) {
 	 * Returns a list of classes that inherit from the given class.
 	 * The resulting array includes the base class passed
 	 * through the $class parameter as the first array value.
-	 * 
+	 *
 	 * Example usage:
 	 * <code>
 	 * ClassInfo::subclassesFor('BaseClass');
 	 * 	array(
-	 * 	0 => 'BaseClass',
+	 * 	'BaseClass' => 'BaseClass',
 	 * 	'ChildClass' => 'ChildClass',
 	 * 	'GrandChildClass' => 'GrandChildClass'
 	 * )
 	 * </code>
-	 * 
+	 *
 	 * @param mixed $class string of the classname or instance of the class
 	 * @return array Names of all subclasses as an associative array.
 	 */
 	public static function subclassesFor($class) {
+		//normalise class case
+		if (is_object($class)) {
+			$className = get_class($class);
+		}
+		else {
+			$reflection = new ReflectionClass($class);
```

Comment:
Repeated from above; Refactoring into a normalisation method would be awesome. :)